google_api_lib/tasks.py

```diff
@@ -55,41 +85,57 @@ def transfer_ownership(self, file, template_file_id) -> None:
 
 @shared_task(base=GoogleApiClientTask, bind=True, priority=TaskPriority.HIGH.value)
 def create_google_sheets(self, puzzle_id) -> None:
-    puzzle = Puzzle.objects.select_related("hunt__settings").get(pk=puzzle_id)
-    template_file_id = (
-        puzzle.hunt.settings.google_sheets_template_file_id
-        or settings.GOOGLE_SHEETS_TEMPLATE_FILE_ID
-    )
-    if not template_file_id:
-        logging.warn(
-            f"Cannot create a sheet for {puzzle.name} as we can't find a sheets template"
-        )
-        return
-    destination_folder_id = (
-        puzzle.hunt.settings.google_drive_folder_id
-        or settings.GOOGLE_DRIVE_HUNT_FOLDER_ID
-    )
-
-    response = create_google_sheets_helper(self, puzzle.name, template_file_id)
-
-    sheet_url = response["webViewLink"]
-    add_puzzle_link_to_sheet(puzzle.url, sheet_url)
-    puzzle.sheet = sheet_url
-    puzzle.save()
-
-    transfer_ownership.delay(response, template_file_id)
-
-    if destination_folder_id:
-        move_drive_file.delay(
-            file_id=response["id"], destination_folder_id=destination_folder_id
+    with transaction.atomic():
+        puzzle = Puzzle.objects.select_related("hunt__settings").get(pk=puzzle_id)
+        template_file_id = (
```

Comment:
nit: `template_file_id` is only used in the `else` block below. You could move this assignment inside the `else` block too to keep the declaration/assignment closer to when the var is use.

Reply:
oh wait, no, nvm, you use it in the `post_create_tasks` function